### PR TITLE
feat(dataset): user can add files to an existing dataset

### PR DESCRIPTION
**Pull request by vfried**
This is almost ready, i just need to do final testing....

image

image

### CODE REVIEW

**Comment by vfried** [comment]
Testing available in: https://datasetstest.dev.renku.ch/

File: src/api-client/dataset.js [hunk 1]
```diff
@@ -30,6 +30,39 @@ export default function addDatasetMethods(client) {
...
+    let headers = client.getBasicHeaders();
+    headers.append('Content-Type', 'application/json');
+    headers.append('X-Requested-With', 'XMLHttpRequest');
+
+    return client.clientFetch(`${client.baseUrl}/renku/cache.project_clone`, {
+      method: 'POST',
+      headers: headers,
+      body: JSON.stringify({
+        depth: 1,
+        git_url: `https://dev.renku.ch/gitlab/${projectPathWithNamespace}.git`
```
Comment (ciyer) [inline src/api-client/dataset.js]: I think this line needs to be fixed.

File: src/dataset/Dataset.present.js [hunk 2]
```diff
@@ -184,11 +184,11 @@ export default function DatasetView(props){
     return <Alert color="danger">{fetchError}</Alert>;
   if(dataset === undefined) return <Loader />;
   if(dataset === null)
-    return <Alert color="danger">Error 404: The dataset that was selected doesn't exist or couldn't be accessed.<br /> <br /> If you just created the dataset try <Button color="danger" size="sm" onClick={()=> window.location.reload()}>reloading</Button> the page.</Alert>
+    return <Alert color="danger">Error 404: The dataset that was selected doesn&apos;t exist or couldn&apos;t be accessed.<br /> <br /> If you just created the dataset try <Button color="danger" size="sm" onClick={()=> window.location.reload()}>reloading</Button> the page.</Alert>
```
Comment (ciyer) [inline src/dataset/Dataset.present.js]: Suggestion:
```
<Alert color="danger">The dataset that was selected does not exist or could not be accessed.<br /> <br /> If you just created the dataset, try <Button color="danger" size="sm" onClick={()=> window.location.reload()}>reloading</Button> the page.</Alert>
```

File: src/project/Project-Route.test.js [hunk 3]
```diff
@@ -62,6 +62,21 @@ describe('basic route extraction', () => {
     expect(pathComponents.projectPathWithNamespace).toEqual('namespace/project-name');
     expect(pathComponents.baseUrl).toEqual('/projects/namespace/project-name');
   });
+  it('handles project-with-namespace datasets listing', () => {
```
Comment (ciyer) [inline src/project/Project-Route.test.js]: 😍

File: src/project/Project.js [hunk 4]
```diff
@@ -113,15 +115,17 @@ function splitProjectSubRoute(subUrl) {
   if (comps.length < 2) return result;
 
   result.projectPathWithNamespace = comps.slice(0, 2).join("/");
-  if (comps.length > 2) {
-    // We need to check if we need to accumulate more components into the projectPathWithNamespace
-    result.projectPathWithNamespace = accumulateIntoProjectPath(result.projectPathWithNamespace, comps.slice(2));
+  if (comps.length > 2) {
```
Comment (ciyer) [inline src/project/Project.js]: There is empty space at the end of the line here.

File: src/project/Project.js [hunk 5]
```diff
@@ -113,15 +115,17 @@ function splitProjectSubRoute(subUrl) {
   if (comps.length < 2) return result;
 
   result.projectPathWithNamespace = comps.slice(0, 2).join("/");
-  if (comps.length > 2) {
-    // We need to check if we need to accumulate more components into the projectPathWithNamespace
-    result.projectPathWithNamespace = accumulateIntoProjectPath(result.projectPathWithNamespace, comps.slice(2));
+  if (comps.length > 2) {	
+    // We need to check if we need to accumulate more components into the projectPathWithNamespace
```
Comment (ciyer) [inline src/project/Project.js]: There is empty space at the end of the line here.

File: src/project/Project.js [hunk 6]
```diff
@@ -113,15 +115,17 @@ function splitProjectSubRoute(subUrl) {
   if (comps.length < 2) return result;
 
   result.projectPathWithNamespace = comps.slice(0, 2).join("/");
-  if (comps.length > 2) {
-    // We need to check if we need to accumulate more components into the projectPathWithNamespace
-    result.projectPathWithNamespace = accumulateIntoProjectPath(result.projectPathWithNamespace, comps.slice(2));
+  if (comps.length > 2) {	
+    // We need to check if we need to accumulate more components into the projectPathWithNamespace	
+    result.projectPathWithNamespace = accumulateIntoProjectPath(result.projectPathWithNamespace, comps.slice(2));
```
Comment (ciyer) [inline src/project/Project.js]: There is empty space at the end of the line here.

File: src/utils/formgenerator/fields/FileUploaderInput.js [hunk 7]
```diff
@@ -175,8 +200,28 @@ function FileuploaderInput({ name, label, alert, setInputs, help, disabled=false
     return filteredFiles;
   }
 
-  useEffect(() => {
-  }, []);
+  let getFileStatusComp = (file) => {
+    switch(file.file_status){
+    case FILE_STATUS.ADDED :
+      return <span><FontAwesomeIcon color="var(--success)" icon={faCheckDouble}/> added</span>
```
Comment (ciyer) [inline src/utils/formgenerator/fields/FileUploaderInput.js]: Is this what this should look like?:

image

Comment (ciyer) [inline src/utils/formgenerator/fields/FileUploaderInput.js]: Would it make sense to display this status as "In dataset" (without an icon), and UPLOADED as "Ready to add" (with a single checkmark)?

Comment (vfried) [reply src/utils/formgenerator/fields/FileUploaderInput.js]: I think this is a good idea, it represents better the fact that the file is in the dataset already.

image

**Comment by vfried** [comment]
Thanks for reviewing the PR, the only change that i have left to do is the URL change that i guess is the most important one. I will find out where to get the gitlab project url from and do the change ASAP.

**Comment by vfried** [comment]
It's fixed now, good catch, that would have made the entire thing fail in renkulab :O

**Pull request merged**